integrations/amazon_bedrock/src/haystack_integrations/components/downloaders/s3/s3_downloader.py

```diff
@@ -0,0 +1,405 @@
+# SPDX-FileCopyrightText: 2023-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+from concurrent.futures import ThreadPoolExecutor
+from pathlib import Path
+from typing import Any, Dict, List, Literal, Optional, Union
+from urllib.parse import urlparse
+from uuid import UUID
+
+from botocore.config import Config
+from botocore.exceptions import ClientError
+from haystack import component, default_from_dict, default_to_dict, logging
+from haystack.dataclasses import ByteStream, Document
+from haystack.utils.auth import Secret, deserialize_secrets_inplace
+
+from haystack_integrations.common.amazon_bedrock.errors import AmazonS3ConfigurationError, AmazonS3Error
+from haystack_integrations.common.amazon_bedrock.utils import get_aws_session
+
+logger = logging.getLogger(__name__)
+
+
+@component
+class S3Downloader:
+    """
+    A component for downloading files from S3 to local filesystem with caching and concurrent downloads.
+    Supports filtering by file extensions and returns documents with file paths.
+
+    Usage example:
+    ```python
+    from haystack import Document
+    from haystack.utils import Secret
+    from haystack_integrations.components.downloaders.s3 import S3Downloader
+
+    import os
+    from pathlib import Path
+
+    downloader = S3Downloader(
+        aws_profile_name=Secret.from_token("prof"),
+        aws_region_name=Secret.from_token("eu-central-1"),
+        download_dir=str(Path.cwd() / "s3_cache"),
+        file_extensions=[".json"],
+        sources_target_type="str",
+    )
+
+    docs = [
+        Document(content="", meta={"s3_bucket": "bucket_name", "s3_key": "your_key", "file_name": "file_name"}),
+    ]
+    result = downloader.run(documents=docs)
+    print("Downloaded docs:")
+    for d in result["documents"]:
+        print(f"- {d.meta.get('file_name')} -> {d.meta.get('file_path')}")
+
+    ```
+    """
+
+    max_cache_size: int = 100
+
+    def __init__(
+        self,
+        aws_access_key_id: Optional[Secret] = Secret.from_env_var("AWS_ACCESS_KEY_ID", strict=False),  # noqa: B008
```

Comment:
Let's make this keyword-only
```suggestion
    def __init__(
        self,
        *,
        aws_access_key_id: Optional[Secret] = Secret.from_env_var("AWS_ACCESS_KEY_ID", strict=False),  # noqa: B008
```